[Model]: add some tests for aria model

xffxff opened the pull request:
https://github.com/vllm-project/vllm/pull/10514 added support for aria model, but didn't include any model tests. This PR is intended to add some tests

xffxff committed:
add some tests for aria model

Signed-off-by: xffxff <1247714429@qq.com>

Isotr0py committed:
update aria test

Signed-off-by: Isotr0py <2037008807@qq.com>

Isotr0py committed:
enable image test

Signed-off-by: Isotr0py <2037008807@qq.com>

Isotr0py committed:
add comments

Signed-off-by: Isotr0py <2037008807@qq.com>

Isotr0py commented:
Hey @xffxff, I managed to get an A800 GPU temporarily to run the Aria model, and I have updated the test for you. PTAL!

xffxff committed:
support setting tokenizer mode in model testing

Signed-off-by: xffxff <1247714429@qq.com>

xffxff commented:
> Hey @xffxff, I managed to get an A800 GPU temporarily to run the Aria model, and I have updated the test for you. PTAL!

Thanks! @Isotr0py  I ran the tests locally, but they didn’t pass initially. After investigating, I found that the vllm runner and hf runner use different tokenizer modes. Currently, Aria only supports the slow tokenizer. I added an option to configure the tokenizer mode in our test utilities, and now all the tests pass in my local environment.

However, I noticed that one of the tests failed when attn_implementation='eager' was set. Removing this line resolved the issue, and all the tests passed successfully after that.

Isotr0py commented:
> However, I noticed that one of the tests failed when attn_implementation='eager' was set. Removing this line resolved the issue, and all the tests passed successfully after that.

Yes, this will disable flash-attention on hf_runner and may cause numeric divergence, because I didn't build flash-attn on that machine to save some time. Not sure why the test can pass with eager implementation on my device, but it's OK to use flash-attn here because large models will be skipped on CI device with only 24GB VRAM.

Isotr0py commented on tests/models/decoder_only/vision_language/test_models.py:
```diff
@@ -134,6 +134,29 @@
+        stop_str=["<|im_end|>"],
+        image_size_factors=[(0.10, 0.15)],
+        max_tokens=64,
+        marks=[large_gpu_mark(min_gb=64)],
```
```suggestion
        marks=[
            pytest.mark.skipif(
                not is_flash_attn_2_available(),
                reason="Model needs flash-attn for numeric convergence.",
            ),
            large_gpu_mark(min_gb=64),
        ],
```
We should add a check for flash_attn_2 here, because original flash_attn_2 is optional in vLLM.  You can import `is_flash_attn_2_available` from `transformers.utils`.

xffxff replied:
Done! PTAL

xffxff committed:
Update tests/models/decoder_only/vision_language/test_models.py

Co-authored-by: Isotr0py <2037008807@qq.com>

xffxff committed:
fix import error

Signed-off-by: xffxff <1247714429@qq.com>
